app/assets/javascripts/redux/modules/models/listenSignals.js

```diff
@@ -140,6 +141,16 @@ export const reducer = handleActions({
 
   [LISTEN_SIGNALS_PUT_REQUEST_FAIL]: handlesListenSignalFailResponse,
 
+  [PROMOTION_SIGNAL_POST_REQUEST_SUCCESS]: (state, action) => {
+    const promotionalTweet = _.get(action.payload, 'promotional_tweet', {});
```

Comment:
Not sure what to do here to add Promotional Tweet Id to the nested promotional_tweet list.. without going so deep.